python-threatexchange/threatexchange/threat_updates.py

```diff
@@ -280,9 +283,11 @@ class ThreatUpdatesStore:
     def __init__(
         self,
         privacy_group: int,
+        types: t.Iterable[str] = (),
     ) -> None:
         self.privacy_group = privacy_group
         self.checkpoint = None
+        self.types = types
```

Comment:
(blocking): You actually can't safely do this, because what types you fetched changes the checkpointing information, and opens the possibility of storing a checkpoint that will give you an incorrect understanding of the state.

The reason I removed types from experiment_fetch was for the reason, and the fix, which is storing a big mapping of checkpoints to types, was super complicated.

For now, I would suggest fetching everything, and filtering down to the indicator types you want, OR by leaving types on ThreatUpdatesDelta, and set it manually after construction inside the body of experimental fetch, which is still hacky, but harder to make mistakes with by accident.

Reply:
I think filtering by types is a fairly important feature though I see how the behavior could be confusing as implemented. I think a quick fix could be to store, with the checkpoint, the types used to do fetch and if the types given are different ignore the checkpoint. This wouldn't be the final behavior but would be correct

Reply:
Kept the argument but no longer used in fetcher